src/Ordering.API/Application/Queries/OrderQueries.cs

```diff
@@ -1,85 +1,57 @@
-            new { userId });
-    }
-
-    public async Task<IEnumerable<CardType>> GetCardTypesAsync()
-    {
-        using var connection = dataSource.OpenConnection();
+        return new Order
+        {
+            ordernumber = order.Id,
+            date = order.GetOrderDate(),
+            description = order.GetDescription(),
+            city = order.Address.City,
+            country = order.Address.Country,
+            state = order.Address.State,
+            street = order.Address.Street,
+            zipcode = order.Address.ZipCode,
+            status = order.OrderStatus.Name,
+            total = order.GetTotal(),
+            orderitems = order.OrderItems.Select(oi => new Orderitem
+            {
+                productname = oi.GetOrderItemProductName(),
+                units = oi.GetUnits(),
+                unitprice = (double)oi.GetUnitPrice(),
+                pictureurl = oi.GetPictureUri()
+            }).ToList()
+        };
 
-        return await connection.QueryAsync<CardType>("SELECT * FROM ordering.cardtypes");
     }
 
-    private Order MapOrderItems(dynamic result)
+    public async Task<IEnumerable<OrderSummary>> GetOrdersFromUserAsync(string userId)
     {
-        var order = new Order
-        {
-            ordernumber = result[0].ordernumber,
-            date = result[0].date,
-            status = result[0].status,
-            description = result[0].description,
-            street = result[0].street,
-            city = result[0].city,
-            state = result[0].state,
-            zipcode = result[0].zipcode,
-            country = result[0].country,
-            orderitems = new List<Orderitem>(),
-            total = 0
-        };
-
-        foreach (dynamic item in result)
-        {
-            var orderitem = new Orderitem
+        return await context.Orders
+            .Include(o => o.OrderItems)
+            .Include(o => o.OrderStatus)
+            .Join(context.Buyers,
+                o => o.BuyerId,  
+                b => b.Id,       
+                (order, buyer) => new { order, buyer })
+            .Where(ob => ob.buyer.IdentityGuid == userId)  
+            .Select(ob => new OrderSummary
             {
-                productname = item.productname,
-                units = item.units,
-                unitprice = (double)item.unitprice,
-                pictureurl = item.pictureurl
-            };
-
-            order.total += item.units * item.unitprice;
-            order.orderitems.Add(orderitem);
-        }
-
-        return order;
+                ordernumber = ob.order.Id,
+                date = ob.order.GetOrderDate(),
+                status = ob.order.OrderStatus.Name,
+                total = (double)ob.order.GetTotal()
+            })
+            .ToListAsync();
     }
+
+    public async Task<IEnumerable<CardType>> GetCardTypesAsync() => 
+        await context.CardTypes.Select(c=> new CardType { Id = c.Id, Name = c.Name }).ToListAsync();
```

Comment:
Why the projection to Id and Name only? Should this just return the CareTypes as they are?

Reply:
Yes could be but in the context we use  eShop.Ordering.Domain.AggregatesModel.BuyerAggregate.CardType and this query returns  eShop.Ordering.API.Application.Queries.CardType. So that would change the public API of this query class.

Reply:
I see, OK.